docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,446 @@
+.. _building_the_project_with_dpcpp:
+
+Building the Project with DPC++
+===============================
+
+This page describes building the oneMKL interface library with either the Intel(R) oneAPI DPC++ Compiler or open-source oneAPI DPC++ Compiler.
+For guidance on building the project with AdaptiveCpp, see :ref:`building_the_project_with_adaptivecpp`.
+
+.. _build_setup_with_dpcpp:
+
+Build Setup
+###########
+
+#. 
+   Install Intel(R) oneAPI DPC++ Compiler (select the variant as described in
+   :ref:`Selecting a Compiler<selecting_a_compiler>`).
+
+#. 
+   Clone this project to ``<path to onemkl>``\ , where ``<path to onemkl>``
+   is the root directory of this repository.
+
+#. 
+   Build and install all `required dependencies <https://github.com/oneapi-src/oneMKL?tab=readme-ov-file#software-requirements>`_. 
+
+.. _build_introduction_with_dpcpp:
+
+Introduction
+############
+
+In most cases, building oneMKL is as simple as setting the compiler and selecting the desired backends to build with.
+
+On Linux (see `Building for Windows`_ for building on Windows):
```

Comment:
Maybe we only need to say that Windows build requires Ninja? Otherwise, it would be confusing that we use CMake but need different commands on Linux and Windows.

Reply:
I've noted this inside the *Building for Windows* section in [7921f06](https://github.com/oneapi-src/oneMKL/pull/510/commits/7921f063967bde46bcefc0991231d6fe490ff2c0).